### PR TITLE
Feature/GH-6115

### LINKED ISSUE #6115: Update Permission core unit tests

### PR DESCRIPTION
**Pull request by lohithosur**
Hi Mihal,
Can you review ".../organization/facade/permission.service.spec.ts"? The test fail for getTypes().

Regards,
Lohit

### CODE REVIEW

**Comment by lohithosur** [comment]
Hi Michal,
I have made the second commit.
I have updated the following <*.spec.ts> files with permission types use case:
1) Action
2) Connector
3) Effect
4) Selector
5) Service

I only have issue with Effect. I'm not sure if the error is in the corresponding permission.effect.ts file? What do you think?

Also kindly review, other <*.spec.ts> files and let me know if they are ok.

Thanks & Regards,
Lohit

File: projects/core/src/organization/facade/permission.service.spec.ts [hunk 1]
```diff
@@ -182,4 +184,21 @@ describe('PermissionService', () => {
       );
     });
   });
+
+  describe('get permission types', () => {
+    it('getTypes() should trigger load permission types when they are not present in the store', () => {
```
Comment (Pucek9) [inline projects/core/src/organization/facade/permission.service.spec.ts:194]: need to check also positive scenario
image

**Review by Pucek9 — CHANGES_REQUESTED** [review]
CR done
- [x] Action
- [x] Connector
- [x] Effect (fixed)
- [x] Selector
- [ ] Service (missed case)

File: projects/core/src/organization/facade/permission.service.spec.ts [hunk 2]
```diff
@@ -31,6 +31,11 @@ const permissionList: EntitiesModel<Permission> = {
   pagination,
   sorts,
 };
+const permissionType2: OrderApprovalPermissionType = {
+  code: 'testPermissionTypeCode',
+  name: 'testPermissionTypeName',
+};
+const permissionTypes2: OrderApprovalPermissionType[] = [permissionType2];
```
Comment (Pucek9) [inline projects/core/src/organization/facade/permission.service.spec.ts]:
```suggestion
const permissionType: OrderApprovalPermissionType = {
  code: 'testPermissionTypeCode',
  name: 'testPermissionTypeName',
};
const permissionTypes: OrderApprovalPermissionType[] = [permissionType]
```

File: projects/core/src/organization/facade/permission.service.spec.ts [hunk 3]
```diff
@@ -200,5 +205,23 @@ describe('PermissionService', () => {
...
+      );
+      let permisstionTypes: OrderApprovalPermissionType[];
+      service
+        .getTypes()
+        .subscribe((data) => {
+          permisstionTypes = data;
+        })
+        .unsubscribe();
+
+      expect(permisstionTypes).toEqual(permissionTypes2);
```
Comment (Pucek9) [inline projects/core/src/organization/facade/permission.service.spec.ts]:
```suggestion
        new PermissionActions.LoadPermissionTypesSuccess(permissionTypes)
      );
      let types: OrderApprovalPermissionType[];
      service
        .getTypes()
        .subscribe((data) => {
          types = data;
        })
        .unsubscribe();

      expect(types).toEqual(permissionTypes);
```

**Review by Pucek9 — APPROVED** [review]

**Pull request merged**